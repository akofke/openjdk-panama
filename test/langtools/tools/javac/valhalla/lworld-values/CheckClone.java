/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support clone
 *
 * @compile/fail/ref=CheckClone.out -XDrawDiagnostics CheckClone.java
 */

final __ByValue class CheckClone {
    final __ByValue class InnerValue {
        void foo(InnerValue iv) {
            iv.clone();
            clone();
        }
    }
    void foo(CheckClone v) {
        v.clone();
        clone();
    }
    @Override
    protected Object clone() { return null; }
}