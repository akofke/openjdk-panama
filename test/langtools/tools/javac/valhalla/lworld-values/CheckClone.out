CheckClone.java:11:21: compiler.err.value.does.not.support: clone
CheckClone.java:12:18: compiler.err.value.does.not.support: clone
CheckClone.java:9:17: compiler.err.empty.value.not.yet
CheckClone.java:16:16: compiler.err.value.does.not.support: clone
CheckClone.java:17:14: compiler.err.value.does.not.support: clone
CheckClone.java:8:13: compiler.err.empty.value.not.yet
6 errors
